Dockerfile

```diff
@@ -171,4 +171,5 @@ ENV LD_LIBRARY_PATH=/usr/lib
 # By default container runs with non-root user
 # Choose root user explicitly only where needed, like - node driver
 RUN useradd --uid 1000 --user-group --shell /bin/bash pmem-csi
+RUN dpkg --purge --ignore-depends=xfsprogs python3 python3-minimal libpython3-stdlib libpython3.7-minimal libpython3.7-stdlib python3.7 python3.7-minimal
```

Comment:
Can you add this to the `RUN` command where packages are installed?

As it stands now, we create a layer where Python is included and later another layer where the files are marked as deleted. But that then just increases the size of our image.

You can check that with `docker history <container name>`.

Reply:
yes, can try that.
But seems testing also fails now because TLS test attempts install of nmap, patch and apt complains. We use final production image and modify it for testing. Should we use testing image for such?

Reply:
moved dpkg cmd up in Dockerfile. But I think we still hit that TLS test failure.

Reply:
> Should we use testing image for such?

And keep Python installed in it? That then doesn't help us with getting rid of Python during our image release process because we'll still have to deal with it for the test image.

This https://serverfault.com/questions/250224/how-do-i-get-apt-get-to-ignore-some-dependencies/506993#506993 answer explains how to hack apt to ignore Python permanently. Alternatively, use `equivs-build` to create a fake python package (https://serverfault.com/a/251091).

Reply:
I did not find easy way to accomplish this with apt options, but there is a brute-force way editing dpkg status file directly. I tried this now (raw variant, just single sed without even looking where we edit, but in current state we have exactly one python3 dependency). Have to admit, this starts to turn hackish, so is it worth doing after all...

Reply:
Is the approach with equivs-build better? See https://github.com/intel/pmem-csi/pull/784